tests/shellcheck/run_shellcheck.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# If $top_srcdir has not been set by automake, detect it
+if [ -z "$top_srcdir" ]; then
+    top_srcdir="$(realpath "$(dirname "$0")/../..")"
+fi
+
+source "${top_srcdir}/tests/shellcheck/ignores.sh"
+logfile="${top_srcdir}/tests/shellcheck/shellcheck.log"
+
+# Check if this test can be run
+if ! type shellcheck > /dev/null 2>&1 ; then
+    echo "SKIP - shellcheck must be installed to run it."
+    exit 77
+fi
+
+# Identify which shellcheck version is running to debug fails due to version bump
+shellcheck --version | tee "$logfile"
+rpm -q ShellCheck | tee -a "$logfile"
+echo | tee -a "$logfile"
+
+# Ignore files according to path patterns
+pushd "${top_srcdir}" > /dev/null || return 1
+files=$(git ls-files)
+# shellcheck disable=SC2154  # comes from the file sourced earlier
+for ignore in "${ignore_patterns[@]}" ; do
+    files=$(echo "$files" | grep -v "$ignore")
+done
+
+# Find files that are shell scripts
+files=$(echo "$files" | xargs file | grep "shell script" | awk -F ":" '{print $1}')
+
+# List files being checked
+echo -e "Files to be checked:\n--------------------\n$files\n" | tee -a "$logfile"
+
+# Do the actual linting
+echo -e "Warnings found:\n---------------" | tee -a "$logfile"
+echo "$files" | xargs shellcheck -f gcc | tee -a "$logfile"
```

Comment:
It looks like `shellcheck` returns useful exit codes. Couldn't we just return any nonzero code returned by `shellcheck` instead of parsing results below?

Reply:
`tee` eats the return code. But If I can remove it - see above - then no need for the parsing...